docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,552 @@
+
+These semantics ensure that all `notify_*` methods are idempotent. Clients can safely retry using the original parameters (memo, block height, destination account).
+
+### Error Handling and Refunds (`NotifyError`)
+
+Calls to `notify_*` methods return a `variant { Ok: ...; Err: NotifyError }`. The `NotifyError` type indicates why an operation failed:
+
+```candid
+type NotifyError = variant {
+  Refunded: record { reason : text; block_index : opt nat64 }; // Request was invalid; ICP (minus fees) was refunded (or attempted). `block_index` is the refund transaction index, if successful.
+  InvalidTransaction: text; // Transaction details (e.g., destination, memo) don't match the request, or it was already processed for a different operation.
+  Processing: null; // The specified block_index is currently being processed by another call. Retry shortly.
+  TransactionTooOld: nat64; // The block_index is older than the CMC's cached history (value returned is the minimum valid block_index). Cannot be retried.
+  Other: record { error_code: nat64; description: text; }; // A miscellaneous error occurred.
+};
+```
+
+**Key Points:**
+
+-   **`Refunded`**: This occurs if the initial ICP transfer had an unrecognized memo (automatic refund), or if the canister creation/top-up/minting failed *after* the ICP transfer was validated (e.g., subnet unavailable, deposit cycles failed).
+    -   **Fees:** When a refund occurs, the standard ICP Ledger transfer fee (`10_000` e8s) is always deducted by the Ledger itself. Additionally, the CMC may burn a specific refund fee from the amount before initiating the refund transfer:
+        -   Create Canister Refund Fee: `10_000` e8s (`CREATE_CANISTER_REFUND_FEE`)
+        -   Top Up Refund Fee: `10_000` e8s (`TOP_UP_CANISTER_REFUND_FEE`)
+        -   Mint Cycles Refund Fee: `10_000` e8s (`MINT_CYCLES_REFUND_FEE`)
+    -   The `reason` field provides details. The `block_index` indicates the ledger block of the successful refund transfer, if applicable.
+-   **`InvalidTransaction`**: The provided `block_index` might correspond to a transaction that doesn't match the request (e.g., wrong memo for the operation, wrong destination subaccount) or was already processed under a *different* `notify_*` call.
+-   **`Processing`**: Indicates contention. Simply retrying the *exact same call* after a short delay is appropriate.
+-   **`TransactionTooOld`**: The CMC only keeps a history of recent notifications (`MAX_NOTIFY_HISTORY` entries, currently 1,000,000). Older transactions cannot be processed. The returned value is the oldest `block_index` the CMC *might* still know about.
+-   **`Other`**: Catches various issues, such as failure to fetch the block from the ledger (`FailedToFetchBlock` error code `1`), internal CMC errors, or authorization issues (`Unauthorized` error code `3` if `notify_create_canister` caller != creator and is not NNS Dapp), bad subnet selection (`BadSubnetSelection` error code `4`).
```

Comment:
is internal CMC errror code '2' ? Just noticed that we don't give the code for this case and that we left out the nubmer 2 :D